homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +72,189 @@ def setup(hass, config):
     username = conf.get(CONF_USERNAME)
     password = conf.get(CONF_PASSWORD)
 
-    zapi = ZabbixAPI(url)
+    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
+
+    entities_filter = convert_include_exclude_filter(conf)
+
     try:
-        zapi.login(username, password)
+        zapi = ZabbixAPI(url=url, user=username, password=password)
         _LOGGER.info("Connected to Zabbix API Version %s", zapi.api_version())
     except ZabbixAPIException as login_exception:
         _LOGGER.error("Unable to login to the Zabbix API: %s", login_exception)
         return False
+    except HTTPError as http_error:
+        _LOGGER.error("HTTPError when connecting to Zabbix API: %s", http_error)
+        zapi = None
+        # Continue even though Zabbix API couldn't be initialized. Zabbix sender later may still work.
```

Comment:
We need to `raise PlatformNotReady` in the platform if `zapi` is None, otherwise the platform will never be setup unless we restart home assistant.

Reply:
How will the `zapi` instance recover?

Reply:
Will that retry setting up the component? For SenderThread, zapi and the initial connect to Zabbix are not required. Can we continue setting up the component and then later retry the zapi initialization? I don't have enough knowledge about start-up procedure to find an adequate solution for this.

Reply:
No, the component setup is not retried via `PlatformNotReady`.

Here's how the influxdb integration retries to setup the component:
https://github.com/home-assistant/core/blob/87b1d1b57fb1a78abcb2bb6e9921d750fddc8cac/homeassistant/components/influxdb/__init__.py#L441-L446

Reply:
We should not continue with an incomplete setup integration.

Reply:
I copied the retry solution from influxdb